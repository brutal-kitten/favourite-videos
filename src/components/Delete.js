import React, { Component } from 'react';

class Delete extends Component {

  handleClick = (event) => {
    event.preventDefault();
    this.props.deleteVideo(this.props.id);
  }

  render() {
    return(
      <div className="delete" >
        <button className="buttonDelete" type="button" onClick={(event) => this.handleClick(event)} >
          <span className="glyphicon glyphicon-trash"></span>
        </button>
      </div>
    )
  }
}

export default Delete;
