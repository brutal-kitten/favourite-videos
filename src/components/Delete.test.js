import React from 'react';
import ReactDOM from 'react-dom';
import Delete from './Delete.js';
import TestRenderer from 'react-test-renderer';
import { shallow, mount, render } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

it('should display a button', () => {
  const tree = TestRenderer.create(
    <div className="delete" >
      <button tabIndex='0' className="buttonDelete" type="button" onClick={(event) => this.handleClick(event)} >
        <span className="glyphicon glyphicon-trash"></span>
      </button>
    </div>).toJSON();
  expect(tree).toMatchSnapshot();
});

describe ('shoud call handleclick function after click', () => {
  const mockDeleteVideo = jest.fn();
  const props = {
    deleteVideo: mockDeleteVideo,
    id: "12345"
  };

  const wrapper = shallow(<Delete {...props}/>);

  wrapper.find('button').simulate('click', {
    preventDefault: () => {}
  });

  it('should call deleteVideo function with 1 argument - id', () => {
    expect(mockDeleteVideo.mock.calls.length).toBe(1);
    expect(mockDeleteVideo.mock.calls[0][0]).toBe("12345");
  });
})
