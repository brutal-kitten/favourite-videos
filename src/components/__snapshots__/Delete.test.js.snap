// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display a button 1`] = `
<div
  className="delete"
>
  <button
    className="buttonDelete"
    onClick={[Function]}
    tabIndex="0"
    type="button"
  >
    <span
      className="glyphicon glyphicon-trash"
    />
  </button>
</div>
`;
